services/auth-service/src/main/java/com/sourcefuse/jarc/services/authservice/services/KeycloakFacadeService.java

```diff
@@ -0,0 +1,76 @@
+package com.sourcefuse.jarc.services.authservice.services;
+
+import com.sourcefuse.jarc.services.authservice.payload.keycloak.KeycloakAuthResponse;
+import com.sourcefuse.jarc.services.authservice.payload.keycloak.KeycloakUserDTO;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class KeycloakFacadeService {
```

Comment:
Whats this service meant for ?

Reply:
this service has 2 functions, one is for for fetching the user profile from keycloak using token, 2nd one is for fetching the token using the code provided